includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php

```diff
@@ -0,0 +1,519 @@
+					'public'                    => (int) $is_public,
+					'name'                      => $this->get_campaign_name( $post ),
+					'fromname'                  => $from_name,
+					'fromemail'                 => $from_email,
+					'p[' . $list_id . ']'       => 1,
+					'm[' . $message['id'] . ']' => 1,
+				],
+				$campaign_data
+			);
+			$campaign      = $this->api_request( 'campaign_create', 'POST', [ 'body' => $campaign_data ] );
+			if ( is_wp_error( $campaign ) ) {
+				// Remove hold in case of creation error.
+				delete_post_meta( $post->ID, 'ac_campaign_id' );
+				return $campaign;
+			}
+			update_post_meta( $post->ID, 'ac_campaign_id', $campaign['id'] );
+		}
+		return [
+			'campaign'    => true, // Satisfy JS API.
+			'campaign_id' => $campaign['id'],
+			'message_id'  => $message['id'],
+			'list_id'     => $list_id,
+			'from_email'  => $from_email,
+			'from_name'   => $from_name,
+		];
+	}
+
+	/**
+	 * Update ESP campaign after post save.
+	 *
+	 * @param string  $post_id Numeric ID of the campaign.
+	 * @param WP_Post $post The complete post object.
+	 * @param boolean $update Whether this is an existing post being updated or not.
+	 */
+	public function save( $post_id, $post, $update ) {
+		$status = get_post_status( $post_id );
+		if ( 'trash' === $status ) {
+			return;
+		}
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+		$this->sync( $post );
+	}
+
+	/**
+	 * Send a campaign.
+	 *
+	 * @param string  $new_status New status of the post.
+	 * @param string  $old_status Old status of the post.
+	 * @param WP_Post $post       Post to send.
+	 *
+	 * @throws Exception Error message if sending fails.
+	 */
+	public function send( $new_status, $old_status, $post ) {
+
+		$post_id = $post->ID;
+		
+		// Only run if the current service provider is ActiveCampaign.
+		if ( 'active_campaign' !== get_option( 'newspack_newsletters_service_provider', false ) ) {
+			return;
+		}
+
+		// Only run if changing to publish.
+		if ( 'publish' !== $new_status || 'publish' === $old_status ) {
+			return;
+		}
+
+		if ( ! Newspack_Newsletters::validate_newsletter_id( $post_id ) ) {
+			return;
+		}
+
+		$campaign_id = get_post_meta( $post_id, 'ac_campaign_id', true );
+		$campaigns   = $this->api_request( 'campaign_list', 'GET', [ 'query' => [ 'ids' => $campaign_id ] ] );
+		if ( is_wp_error( $campaigns ) ) {
+			$error = $campaigns;
+		} else {
+			$campaign           = $campaigns[0];
+			$iso_reference_date = new DateTime( $campaign['sdate_iso'] );
+			$schedule_date      = ( new DateTime() )->setTimezone( $iso_reference_date->getTimezone() );
+			$send_result        = $this->api_request(
+				'campaign_status',
+				'GET',
+				[
+					'query' => [
+						'id'     => $campaign_id,
+						'status' => 1,
+						'sdate'  => $schedule_date->format( 'Y-m-d H:i:s' ),
+					],
+				] 
+			);
+			if ( is_wp_error( $send_result ) ) {
+				$error = $send_result;
+			}
+		}
+
+		if ( $error ) {
```

Comment:
`$error` might not be defined (if everything went fine), causing a `Undefined variable: error` PHP notice.

Reply:
Fixed by assigning the variable earlier in 79eea75b82fa54b314f5f05d091c752dd160573c